### PR TITLE
Add NPC retreat, no investigate, and investigate limit zones

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Features "Add NPC retreat, no investigate, and investigate limit zones"```

#### Purpose of change
Part of #29333 
Add 3 new zones for friendly NPCs:
* Retreat zones give the NPC a preferred location to retreat to if the zone is within 60 tiles and the NPC is fleeing threats or a fire
* No investigate noises prevent the NPC from investigating sounds coming from inside the zone.
* Investigate limit zones prevent the NPC from investigating sounds coming from outside the zone.

#### Describe the solution
Add two new functions to the zone manager: get_nearest returns the nearest point of a zone type within a specified distance, and has_defined returns true if there is a cached zone type.

In npcmove.cpp, good_escape_dir will attempt to path to the nearest retreat zone in preference to using the threat map.

in npctalk.cpp, handle_sounds will skip investigating sounds that come from inside a no investigate zone, or if a investigate limit zone is defined, from sounds coming from outside of it.

Update the in-dialogue tutorial to include the new features.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
I wonder if there might be more appropriate names for "no investigate" and "investigate limit". Those aren't very self explanatory.

**Comment by mlangsdorf** [comment]
I am open to suggestions for better names.  They make sense to me and all zone names are kind of cryptic, honestly.

**Comment by I-am-Erk** [comment]
Yeah, I'm struggling to think of anything better at the moment too. If it comes to mind I'll let you know

File: src/clzones.cpp [hunk 1]
```diff
@@ -33,6 +33,15 @@ zone_manager::zone_manager()
     types.emplace( zone_type_id( "NO_NPC_PICKUP" ),
                    zone_type( translate_marker( "No NPC Pickup" ),
                               translate_marker( "Friendly NPCs don't pickup items inside the zone." ) ) );
+    types.emplace( zone_type_id( "NPC_RETREAT" ),
+                   zone_type( translate_marker( "NPC Retreat" ),
+                              translate_marker( "Friendly NPCs will attempt to retreat to this zone if it is nearby." ) ) );
```
Comment (Night-Pryanik) [inline src/clzones.cpp]: "Nearby" is rather vague. Maybe we should tell the exact distance?

**Comment by I-am-Erk** [comment]
No investigate zone - how about "ignore sounds" zone?

Investigate limits - I think this one is probably okay if 'no investigate' is changed since it makes them less similar sounding. However I sort of vaguely think "investigat*ion* limits" sounds a little clearer in intent.

**Comment by RafeHaab** [comment]
"Don't investigate"

"Investigate within reason/limits"

**Pull request merged**